src/Microsoft.AspNetCore.Razor.Language/DefaultRazorCSharpLoweringPhase.cs

```diff
@@ -60,6 +60,7 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
 
             var diagnostics = new List<RazorDiagnostic>();
             diagnostics.AddRange(syntaxTree.Diagnostics);
+            diagnostics.AddRange(irDocument.GetAllDiagnostics());
```

Comment:
I was debating this and was thinking it was kind of weird how we represent the entire worlds errors on the CSharpDocument to begin with. I feel that the only errors that make sense on the CSharpDocument itself are those that were created during lowering. 

How do you feel about adding a code document extension that brought together all the errors? Or even having an extra extension that built a final "Razor result" to being processed?